File: test47_out/test47/Dispatcher/Dispatcher.js

/*
 * ======== MODULE test47.Dispatcher ========
 */

var $$c = function() {
    const Dispatcher = {};
    Dispatcher.Dispatcher = Dispatcher
    Dispatcher.$name = 'test47.Dispatcher';
    Dispatcher.pollen$used = 0;
    Dispatcher.Ev = $units['test47.Ev'];
    Dispatcher.EvQueue = $units['test47.EvQueue'];
    Dispatcher.post = new $$Ref('test47_Dispatcher_post__E');
    Dispatcher.run = new $$Ref('test47_Dispatcher_run__E');
    Dispatcher.targetInit = new $$Ref('test47_Dispatcher_targetInit__I');
    Dispatcher.$$hostInit = function() {
        var $$text = '';
        Dispatcher.myEventQueue = new $units['test47.EvQueue'].EvQueue('test47_Dispatcher_myEventQueue__V'); Dispatcher.myEventQueue.new_host(10);
        return $$text;
    }
    Dispatcher.pollen__uses__ = function() {
        $$bind($units['test47.Ev'], 'pollen$used', true);
        $$bind($units['test47.EvQueue'], 'pollen$used', true);
    }
    Dispatcher.$$privateInit = function() {
    }
    return Dispatcher;
}
var $$u = $$c();
$units['test47.Dispatcher'] = $$u;
$units.push($$u);

File: test47_out/test47/Dispatcher/Dispatcher.c

/*
 * ======== string literals ========
 */

static const char test47_Dispatcher__s__fileName[] = "Dispatcher.p";
static const char test47_Dispatcher__s__0[] = "Dispatcher";

/*
 * ======== function definitions ========
 */

void test47_Dispatcher_post__E( struct test47_Ev*  e ) {
    test47_EvQueue_EvQueue_add__E(&(test47_Dispatcher_myEventQueue__V), (e));
}

void test47_Dispatcher_run__E() {
    test47_Ev_ e;
    while (true) {
        e = test47_EvQueue_EvQueue_remove__E(&(test47_Dispatcher_myEventQueue__V));
        if ((e) != null) {
            test47_Ev_Ev_handle__E(e, test47_Ev_Ev_getData__E(e));
            break;
        }
    }
}

void test47_Dispatcher_targetInit__I() {
}

